consumer/consumer.py

```diff
@@ -0,0 +1,345 @@
+from typing import List
+
+import glob
+import json
+
+from io import BytesIO
+from minio import Minio
+from minio.error import S3Error
+
+from confluent_kafka import Consumer
+from google.protobuf.json_format import MessageToJson
+
+from config.config import ConfigParser, KafkaConsumerProperties, get_config_path
+from events_registry.events_registry import events_mapping
+from events_registry.key_manager import ProducerKeyManager
+
+CONFIG_FILE_PATH = get_config_path()
+
+# ==================================== #
+# (1) Create consumer instance
+# ==================================== #
+
+def create_kafka_consumer() -> Consumer:
+    config_parser = ConfigParser(CONFIG_FILE_PATH)
+    kafka_consumer_config_dict = config_parser.get_consumer_config()
+    kafka_consumer_config = KafkaConsumerProperties(**kafka_consumer_config_dict)
+    print("Kafka consumer config:", kafka_consumer_config)
+    return Consumer(kafka_consumer_config.model_dump(by_alias=True))
+
+
+# TODO: Do we need message received report? (by analogy with message delivery for producer). Can be done with Prometheus
+
+# ==================================== #
+# (2) Parse messages from kafka
+# ==================================== #
+
+def parse_message(message):
+    event_type = ProducerKeyManager(producer_key=message.key().decode("utf-8")).get_event_type_from_key()
+    # print("\nEvent type: ", event_type)
+
+    binary_value = message.value()
+    # print("\nValue:", binary_value)
+
+    if event_type in events_mapping:
+        event_class = events_mapping.get(event_type)
+        event = event_class()
+        event.ParseFromString(binary_value)
+        event_json_string = MessageToJson(event)
+        event_json_data = json.loads(event_json_string)
+        print(f"Event class data type: {type(event_json_data)}, data: \n{event_json_data}")
+        return event_json_data
+
+
+# ==================================== #
+# (3) Set up stuff for Minio
+# ==================================== #
+
+def create_minio_client() -> Minio: 
+    # TODO: Move credentials from here to stg config
+    minio_client = Minio(
+                "localhost:9000", 
+                # following keys are from the environment variables in docker-compose.yaml 
+                access_key="minio_user",  
+                secret_key="minio_password",  
+                secure=False  # Set to True if using HTTPS
+            )
+    return minio_client
+
+def create_bucket(bucket_name: str, minio_client: Minio) -> None:
+    try:
+        if not minio_client.bucket_exists(bucket_name):
+            minio_client.make_bucket(bucket_name)
+            print(f"Bucket '{bucket_name}' created successfully.")
+        else:
+            print(f"Bucket '{bucket_name}' already exists.")
+    except S3Error as exc:
+        print("Error occurred: ", exc)
+
+# ==================================== #
+# (3) Set up FileWriters
+# ==================================== #
+
+class FileWriter(abc.ABC):
```

Comment:
Usually, for abs. classes we use naming like `FileWriterBase` or smth like that, to distinguish it from the real classes